src/app/core/services/api.service.ts

```diff
@@ -71,12 +62,23 @@ export class ApiService {
     },
     VolumeDataRequest:{
       apiCall:{
-        protocol:"rest",
-        version:"1.0",
-        operation: "get",
-        namespace: "storage/volume/",
+        protocol:"websocket",
+        version:"2.0",
+        namespace: "pool.dataset.query",
+        args: [],
         responseEvent: "VolumeData"
-      }
+      },
+      preProcessor(def:ApiCall){
+        let redef = Object.assign({}, def);
+
+        const queryFilters = [
+          ["name","~", "^[^\/]+$" ], // Root datasets only
+        ];
+
+        const params = [queryFilters];
+        redef.args = params;
+        return redef;
+      },
```

Comment:
I think this could be shortened without sacrificing readability:
```ts
{
  preProcessor(def: ApiCall) {
    return { args: [[["name", "~", "^[^\/]+$"]]], ...def }; // Root datasets only
  }
}
```

Reply:
Yes it could be a little less verbose. I dislike too many nested arrays though. I changed it to the following...

```
      preProcessor(def:ApiCall){
        const queryFilters = [
          ["name","~", "^[^\/]+$" ], // Root datasets only
        ];

        return { args: [queryFilters], ...def };
      },

```